src/tables.ts

```diff
@@ -586,6 +718,41 @@ export class Tables {
     return clefs[clef];
   }
 
+  /**
+   * Use the provided key to look up a value in this font's metrics file (e.g., bravuraMetrics.ts, petalumaMetrics.ts).
+   * @param key is a string separated by periods (e.g., stave.endPaddingMax, clef.lineCount.'5'.shiftY).
+   * @param defaultValue is returned if the lookup fails.
+   * @returns the retrieved value (or `defaultValue` if the lookup fails).
+   */
+  // eslint-disable-next-line
+  static lookupMetric(key: string, defaultValue?: Record<string, any> | number): any {
+    const keyParts = key.split('.');
+
+    let currObj;
+
+    while (!currObj && keyParts.length) {
+      // Start with the top level font metrics object, and keep looking deeper into the object (via each part of the period-delimited key).
+      currObj = CommonMetrics;
+      for (let i = 0; i < keyParts.length; i++) {
+        const keyPart = keyParts[i];
+        const value = currObj[keyPart] as any;
+        if (value === undefined) {
+          // If the key lookup fails, we fall back to undefined.
+          currObj = undefined;
+          break;
+        }
+        // The most recent lookup succeeded, so we drill deeper into the object.
+        currObj = value;
+      }
+      if (keyParts.length > 1) {
+        keyParts[keyParts.length - 2] = keyParts[keyParts.length - 1];
+      }
+      keyParts.pop();
+    }
+
+    // Return the retrieved or default value
+    return currObj ? currObj : defaultValue;
+  }
   /**
```

Comment:
There should be an empty line after the end of `lookupMetric()` and before this comment block.

Reply:
done